onnx/defs/traditionalml/defs.cc

```diff
@@ -302,57 +302,122 @@ ONNX_ML_OPERATOR_SET_SCHEMA(
             AttributeProto::INT,
             static_cast<int64_t>(0)));
 
-static const char* LabelEncoder_ver1_doc = R"DOC(
-    Converts strings to integers and vice versa.<br>
-    If the string default value is set, it will convert integers to strings.
-    If the int default value is set, it will convert strings to integers.<br>
-    Each operator converts either integers to strings or strings to integers, depending 
-    on which default value attribute is provided. Only one default value attribute
-    should be defined.<br>
-    When converting from integers to strings, the string is fetched from the
-    'classes_strings' list, by simple indexing.<br>
-    When converting from strings to integers, the string is looked up in the list
-    and the index at which it is found is used as the converted value.
+static const char* LabelEncoder_ver2_doc = R"DOC(
+    Encode values as integers or map integers to other values.<br>
+    Label Encoder has two operation modes. The first mode, named
+    indexing mode, only works if and only if the input is an integer
+    tensor. Indexing mode uses each input element as an index to retrieve
+    the associated value in the specified 'classes_*' field. For example, if
+    input is [1, 0, 2, 0], 'classes_floats' is [5.5, 6.6], and
+    'default_float' is -8.7, the expected output would be [6.6, 5.5, -8.7,
+    5.5]. Let's consider another example. Assume that input is [1, 2, 2, 0]
+    and 'classes_strings' is ["Sally", "Dori", "Amy"]. The output would be
+    ["Dori", "Amy", "Amy", "Sally"]. The other mode maps non-integer types
+    to integer types, called loop-up mode. Each element in the input
+    tensor is used as a position index to the associated 'classes_*'
+    attribute. Let ["A", "B", "B", "C"] be the input, 'classes_strings' be
+    ["C", "B"], 'default_int64' be -99. Then, the corresponding output
+    should be [-99, 1, 1, 0] because "C" and "B" are respectively the first
+    (indexed by 0) and the second (indexed by 1) elements in
+    'classes_strings' while "A" cannot be found in 'classes_strings'.<br>
+    Under loop-up mode, only default_int64 can be set because it is the
+    integer value used whenever indexing fails (e.g., when an integer
+    larger than the size of 'classes_* is given). In the meanwhile, input
+    type is identical to the element type of the specified 'classes_' while
+    output is always an integer tensor. For indexing mode, the input type
+    is always integer tensor and the output type is identical to the
+    specified specified 'classes_*'. The 'default_*' field is the output
+    value whenever loop-up fails for an element. Note that only one of
```

Comment:
look-up? It looks like look-up and indexing may be mixed up in the above few lines.